### ISSUE TITLE
 Sonar does not display tests and code coverage 

### ISSUE THREAD

**Issue by issemLenoil** [Backelite/sonar-swift#102]
I am using **Sonarqube 6.5** and **sonar-swift 0.3.5**.

My tests have structure like this:
> -- **MyAppTests**
> ------ MyExampleTest.swift
> ------ **FolderWithOtherTests**
> ---------- MyClassTest.swift
> ------ **AnotherFolderWithTests**
> ---------- AnotherClassTest.swift

In *sonar-project.properties* I set the following path to test directories:
`sonar.tests=MyAppTests`

After running **run-sonar-swift.sh**, I get logs:
> WARN: Resource not found: MyAppTests.MyClassTest
> WARN: Resource not found: MyAppTests.AnotherClassTest

why are *MyClassTest*, *AnotherClassTest* searched directly in the *MyAppTests* folder?

Finally, on the *Sonarqube dashboard*, in section *Measures/Coverage*, only test *MyExampleTest.swift* is visible.
Percentage unit tests coverage is not shown.

Sonar reports *coverage.xml* and *TEST-report.xml* are generated properly.

Has anyone got an idea what could be the cause?
How to solve this problem?

**Comment by singh88** [comment]
I am getting the same error is there any specific reason for this?

**Comment by singh88** [comment]
Has anyone any suggestions on this?

**Comment by eduardourso** [comment]
I'm also facing this issue, +1. @issemLenoil or @singh88 have you found a solution for your problem?

**Comment by singh88** [comment]
No @eduardourso

**Comment by sorinant** [comment]
In order to fix this issue I had to move all my tests in the root folder of the tests project, in your example MyAppTests. I think the plugin or sonar don't recursively go through the tests folder in order to detect all of them.

**Comment by singh88** [comment]
I found the solution for this problem @eduardourso. I was using [fastlane](https://github.com/Backelite/sonar-swift/blob/develop/docs/sonarqube-fastlane.md) configuration. After running this I found that under `reports` directory coverage reports was getting generated with name `cobertura.xml`. So, I have modified the Fastlane configuration script to include a step to rename cobertura.xml to coverage-swift.xml (this string is referenced by sonar-swift code base [here](https://github.com/Backelite/sonar-swift/blob/a404d5f64ae74dcf3bcad42c95a3df083c870fa6/sonar-swift-plugin/src/main/java/org/sonar/plugins/swift/coverage/SwiftCoberturaSensor.java#L39) and that solved my problem.
```
lane :metrics do
    scan(scheme: "[SCHEME]", code_coverage: true, derived_data_path: "./DerivedData", output_directory: "./sonar-reports")
    slather(cobertura_xml: true, jenkins: true, scheme: "[SCHEME]", build_directory: "./DerivedData", output_directory: "./sonar-reports", proj: "./[PROJECT].xcodeproj")
    sh("cd .. && lizard ./[SOURCE_FOLDER] -l swift --xml > ./sonar-reports/lizard-report.xml")
    swiftlint(output_file: "./sonar-reports/swiftlint.txt", ignore_exit_status: true)
    sh("cd .. && mv sonar-reports/cobertura.xml sonar-reports/coverage-swift.xml")
    sonar
end
```

**Comment by eduardourso** [comment]
@singh88 awesome! thanks for sharing it, I also checked in the PRs and it looks like they fixed it here #88, I'll give it a go and update my `run-sonar-swift.sh`.

**Comment by qoomon** [comment]
just use the develop branch version of [run-sonar-swift.sh](https://github.com/Backelite/sonar-swift/blob/develop/sonar-swift-plugin/src/main/shell/run-sonar-swift.sh)
